OMCompiler/Compiler/BackEnd/RemoveSimpleEquations.mo

```diff
@@ -192,6 +194,79 @@ algorithm
   end if;
 end removeVerySimpleEquations;
 
+public function fixAliasVarsVariablity
+" This is a workaround for #5673
+  This module traverses all alias variables and double-checks if the referenced alias variables variablitly
+  is parameter or not and changes variablilty of alias variable to BackendDAE.PARAM()."
+  input BackendDAE.BackendDAE inDAE;
+  output BackendDAE.BackendDAE outDAE;
+protected
+  BackendDAE.Variables aliasVars,systvars,globalKnownVars;
+  DAE.Exp binding;
+  DAE.ComponentRef cr;
+  Boolean param;
+  BackendDAE.Shared shared;
+  list<BackendDAE.Var> referencevar,aliasVarList = {},knownVarList={};
+algorithm
+  aliasVars := BackendDAEUtil.getAliasVars(inDAE);
+  systvars := getVarsFromEqSystem(inDAE.eqs);
+  //BackendDump.dumpVariables(aliasVars,"aliasVariable-Actual");
+  for var in BackendVariable.varList(aliasVars) loop
+    binding := BackendVariable.varBindExp(var);
+    {cr} := Expression.getAllCrefs(binding);
+    // look up for the reference variable variablity
+    if BackendDAEUtil.containAnyVar({cr}, systvars) then  //check in EqSystem.OrderedVars
+      (referencevar,_) := BackendVariable.getVar(cr,systvars);
+    elseif BackendDAEUtil.containAnyVar({cr}, inDAE.shared.globalKnownVars) then //check in shared.globalKnownVars
+      (referencevar,_) := BackendVariable.getVar(cr,inDAE.shared.globalKnownVars);
+    else
+      // some other cases, example variables in shared.externalObjects, and shared.localKnownVars do not handle this case for now
+      referencevar :={};
+    end if;
+    param := checkVariability(referencevar);
+    if (param) then
+      // remove the variable from AliasVarList and add it to GlobalknownVarlist
+      knownVarList := BackendVariable.setVarKind(var,BackendDAE.PARAM()) :: knownVarList;
+      knownVarList := BackendVariable.setVarFixed(var,true) :: knownVarList;
+      //BackendDump.dumpVarList(referencevar,"matchedparamvar");
+    else
+      aliasVarList := var :: aliasVarList;
+    end if;
+    //print("Alias var : " + ComponentReference.crefStr(var.varName) + "==>" + ComponentReference.crefStr(cr) + "\n");
+  end for;
+  //BackendDump.dumpVarList(aliasVarList,"AfterChangingParameters");
+
+  // add the parameter dependent alias vars to Global known Vars after removing from aliasVarList
+  globalKnownVars := BackendVariable.addVars(knownVarList,inDAE.shared.globalKnownVars);
```

Comment:
For testing purpose, it might be helpful to add the variables to the end of the already given global known vars. What do you think?

Reply:
Yes this can be done